server.c

```diff
@@ -171,7 +172,55 @@ void read_into_buf(int sock, Buffer *buffer) {
 	buffer->buf[buffer->curr_size] = 0;  // add null char
 }
 
-int main(int argc, char **argv) {
+bool check_err(char* str, char* end, int n) {
+	if (str == end) {
+		// message error (message empty)
+		return false;
+	}
+	bool first_two = (*str == 'M') && isdigit(str+1) && *(str+1) == n+1;

```

Comment:
not sure about isdigit(str+1), isnt str+1 a pointer?

*(str+1) == n+1 wont work. comparing int and char